project/__init__.py

```diff
@@ -25,6 +25,7 @@ class BaseProject(object):
         'Gemfile': 'bundle update',
         'package.json': 'npm update',
         'go.mod': 'go get -u all',
+        'yarn.lock': 'yarn upgrade'
```

Comment:
Can you explain why this is needed?

As I understand it, we want to update lock files whenever dependencies have changed. For JS projects, these dependencies are declared in `package.json`. So a change in `yarn.lock` means that a change in `package.json` already occurred. This seems to already be handled by `'package.json': 'npm update'` (which is overwritten in projects using Yarn if I understood correctly.) So why do we need this line?
Also, do some projects use npm and others use Yarn? Why don’t we do `'package.json': 'yarn upgrade'` and avoid `npm` entirely? This way we wouldn’t have to overwrite `package.json` in `updateCommands` for some projects and not for others.

Also, unless you use `yarn install --frozen-lockfiles`, Yarn will update `yarn.lock` if needed (i.e. if the information in the lock files aren't enough to satisfy all the dependencies constraints declared in `package.json`. This includes new dependencies, dependencies deletion, change in version numbers.) So `yarn install`ing a project in which a `package.json` file has changed should update the `yarn.lock` file at the same time. That would make the `'pakcage.json'` updateCommand useless I think?

I might lack some background information/implicit knowledge on the project to understand these choices.